samples/ck-rx65n/basic-sample/src/iotconnect_app.c

```diff
@@ -0,0 +1,267 @@
+        free((void*) ack);
+    }
+    if (needs_ota_commit) {
+        printf("Waiting for ack to be sent by the network\r\n.,,");
+        tx_thread_sleep(5 * NX_IP_PERIODIC_RATE);
+        UINT status = iotc_ota_fw_apply();
+        if (status) {
+            printf("Failed to apply firmware! Error was: %d\r\n", status);
+        }
+    }
+}
+
+static void command_status(IotclEventData data, bool status, const char *command_name, const char *message) {
+    const char *ack = iotcl_create_ack_string_and_destroy_event(data, status, message);
+    printf("command: %s status=%s: %s\r\n", command_name, status ? "OK" : "Failed", message);
+    printf("Sent CMD ack: %s\r\n", ack);
+    iotconnect_sdk_send_packet(ack);
+    free((void*) ack);
+}
+
+static void on_command(IotclEventData data) {
+	char* context = NULL;
+    char *command = iotcl_clone_command(data);
+    if (NULL != command) {
+    	char* token = strtok_r(command, " ", &context);
+    	if(!strncmp(token, "led", 3)) {
+    		token = strtok_r(NULL, " ", &context);
+    		RX65N_LED_STATE state = (token[0] == '1') ||
+    								(strncmp(token, "true", 4) == 0 ) ?
+    								ON : OFF;
+    		set_led(LED2, state);
+    		set_led(LED1, state);
+    		command_status(data, true, command, "LED set");
+    	}
+    	else {
+    		command_status(data, false, command, "Not implemented");
+    	}
+        free((void*) command);
+    } else {
+        command_status(data, false, "?", "Internal error");
+    }
+}
+
+static void on_connection_status(IotConnectConnectionStatus status) {
+    // Add your own status handling
+    switch (status) {
+    case MQTT_CONNECTED:
+        printf("IoTConnect Client Connected\r\n");
+        break;
+    case MQTT_DISCONNECTED:
+        printf("IoTConnect Client Disconnected\r\n");
+        break;
+    default:
+        printf("IoTConnect Client ERROR\r\n");
+        break;
+    }
+    if (NULL != auth_driver_context) {
+    	release_sw_der_auth_driver(auth_driver_context);
+    }
+}
+
+static void publish_telemetry() {
+    const char *str;
+    IotclMessageHandle msg = iotcl_telemetry_create(iotconnect_sdk_get_lib_config());
+
+    // Optional. The first time you create a data point, the current timestamp will be automatically added
+    // TelemetryAddWith* calls are only required if sending multiple data points in one packet.
+    iotcl_telemetry_add_with_iso_time(msg, iotcl_iso_timestamp_now());
+    iotcl_telemetry_set_string(msg, "version", APP_VERSION);
+//   iotcl_telemetry_set_number(msg, "cpu", 3.123); // test floating point numbers
+    // random number 0-100, cast to int so that it removes decimals in json
+    iotcl_telemetry_set_number(msg, "random", (int)((double)rand() / (double)RAND_MAX * 100.0));
+
+    iotcl_telemetry_set_bool(msg, "button", read_user_switch());
+
+//    sensors_add_telemetry(msg);
+    str = iotcl_create_serialized_string(msg, false);
+    iotcl_telemetry_destroy(msg);
+    printf("Sending: %s\r\n", str);

```

Comment:
Move printf inside the `if (NULL != str) `{ block

Reply:
@nmatthews-witekio  And what do we do if it indeed is null? Null is not returned ever bu that function unless user messed up something or we ran out of RAM. Probably better to crash than silently ignore it.